litsearch/getInternalData.py

```diff
@@ -0,0 +1,160 @@
+import pandas as pd
+import requests
+import json
+import base64
+
+
+# Issues Helper Functions
+def getIssuesFromAPI():
+    """ Gets all the issues and pull-requests (GH treats PRs like issues in the api)
+    Needs to use pagination because of 100 per-request api limit
+    """
+    issues = []
+    pageNumber = 1
+    numberOfIssuesReturned = 1
+    while numberOfIssuesReturned != 0:
+        issuesResponse = requests.get(
+            "https://api.github.com/repos/greenelab/covid19-review/issues?state=all&per_page=50&page=" +
+            str(pageNumber))
+        issues_page = json.loads(issuesResponse.text)
+        issues = issues + issues_page
+        numberOfIssuesReturned = len(issues_page)
+        pageNumber += 1
+    return issues
+
+
+def getCitationFromIssue(issue):
+    """ Gets the citation from the github issue assuming the issue follows the New Paper issue template
+    Citation is typically a DOI but could be something else
+    """
+    try:
+        if "\nCitation: " in issue["body"]:
+            citation = issue["body"].split("\nCitation: ")[1].split(" ")[0]
+        else:
+            afterDOI = issue["body"].split("DOI:")[1]
+            citation = afterDOI.split(" ")[0]
+            if citation == "":
+                citation = afterDOI.split(" ")[1]
+        if "\r\n" in citation:
+            citation = citation.split("\r\n")[0]
+        return citation
+
+    except:
+        print(
+            "the citation could not be automatically extracted from the following issue: \n",
+            issue["title"])
+        return "unknown"
+
+
+def getPaperTitleFromIssue(issue):
```

Comment:
If we are able to extract the citation, we could again use Manubot to get the title.

Reply:
This could be helpful. Would I call `manubot cite` via a subprocess or is there other python integration?

Reply:
@rdvelazquez the manubot Python package has some importable functions you can use instead of calling the `manubot cite` command as a subprocess.

The best function is [`citekey_to_csl_item`](https://github.com/manubot/manubot/blob/a4c826ceb954635810ba204a2393dce6fa7a5d5a/manubot/cite/citekey.py#L278).  This takes in a Manubot-style citekey like `doi:10.1101/2020.04.08.20054551`.  It works for all valid citekeys (e.g. PMIDs, URLs) so if you ever needed to expand beyond DOIs you could use the same function.

The [test cases](https://github.com/manubot/manubot/blob/a4c826ceb954635810ba204a2393dce6fa7a5d5a/manubot/cite/tests/test_citekey_api.py) and (to a lesser extent) [CSL Item docs](https://manubot.github.io/manubot/reference/manubot/cite/csl_item/) will give you a sense of how to use the returned CSL Item object.  The [DOI tests](https://github.com/manubot/manubot/blob/a4c826ceb954635810ba204a2393dce6fa7a5d5a/manubot/cite/tests/test_doi.py) show more ways to interact with the CSL Item you'll get back.

If the citation information in the issue is well-formatted (not a safe assumption), something like this should work:
```python
citekey = getCitationFromIssue(issue)
# citekey is now a string like doi:10.1038/s41579-018-0118-9
csl_item = citekey_to_csl_item(citekey)
title = csl_item["title"]
# if ever needed at a later date for cross-referencing (not all CSL Items have valid PMID of course)
pmid = csl_item["PMID"]
```

Reply:
This is great to know about!

Reply:
Very cool; that's exactly what I was looking for. I'll give `citekey_to_csl_item` a try.